### PR TITLE
Abocquet/start frontend

### PR DESCRIPTION
**Pull request by abocquet**

### CODE REVIEW

File: frontend/src/Navbar.tsx [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+
+import {
+  NavbarToggler,
+  UncontrolledDropdown,
+  DropdownToggle,
+  DropdownMenu,
+  DropdownItem } from 'reactstrap';
+
+function Navbar() {
+	const [isOpen, setIsOpen] = useState(true);
```
Comment (Piwakk) [inline frontend/src/Navbar.tsx]: `useState<boolean>` is a bit better

File: frontend/src/Navbar.tsx [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+import {Link} from "react-router-dom";
+
+import {
+  NavbarToggler,
+  UncontrolledDropdown,
+  DropdownToggle,
+  DropdownMenu,
+  DropdownItem } from 'reactstrap';
+
+function Navbar() {
```
Comment (Piwakk) [inline frontend/src/Navbar.tsx]: No props?

File: frontend/src/models/associations/election.ts [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+
+export interface Choice {
+    id: string;
+    election: Election;
+}
+
+export interface Ballot {
+    id: string;
+    election: Election;
+    choices: Choice;
```
Comment (Piwakk) [inline frontend/src/models/associations/election.ts]: Choice[]

Comment (Piwakk) [reply frontend/src/models/associations/election.ts]: or `choice`

File: frontend/src/models/associations/election.ts [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+}
+
+export interface Election {
+    id: string;
+    association: string;
+    name: string;
+    choices: Choice[];
+    registered_voters: User[];
```
Comment (Piwakk) [inline frontend/src/models/associations/election.ts]: Snake or Camel case?

Snake will make it easier to work with the JSON (as the JSON fields are snake case). Nevertheless, I think we should follow Camel to suit the TS conventions.

In any case, we have to be consistent; it's `Association.myRole` a bit above

File: frontend/src/models/associations/election.ts [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+    starts_at: Date;
+    ends_at: Date;
+    max_choices_per_ballot: string;
```
Comment (Piwakk) [inline frontend/src/models/associations/election.ts]: Not a `number`?

File: frontend/src/models/associations/election.ts [hunk 6]
```diff
@@ -0,0 +1,23 @@
...
+
+export interface Ballot {
+    id: string;
+    election: Election;
+    choices: Choice;
+}
+
+export interface Election {
+    id: string;
+    association: string;
```
Comment (Piwakk) [inline frontend/src/models/associations/election.ts]: `Association`?

File: frontend/src/models/associations/election.ts (hunk 4 above)
Comment (abocquet) [reply frontend/src/models/associations/election.ts]: Camel

File: frontend/src/models/polls.ts [hunk 7]
```diff
@@ -0,0 +1,28 @@
+import {User} from "./user";
+
+export enum PollState {
+    ACCEPTED = "ACCEPTED",
```
Comment (Piwakk) [inline frontend/src/models/polls.ts]: Decide if it's either `PollState.ACCEPTED` or `PageType.News` (upper case or not).
TypeScript documentation only uppers the first letter

File: frontend/src/models/repartitions.ts [hunk 8]
```diff
@@ -0,0 +1,42 @@
+import { User } from "./user";
+
+export enum CampaignStatus {
+    CLOSED = "CLOSED",
```
Comment (Piwakk) [inline frontend/src/models/repartitions.ts]: Same here for the upper letters

File: frontend/src/models/repartitions.ts [hunk 9]
```diff
@@ -0,0 +1,42 @@
...
+export interface Group {
+    id: string;
+    proposition: Proposition;
+    campaign: Campaign;
+    users: User[];
+}
+
+export interface Wish {
+    proposition: string;
+    rank: string;
```
Comment (Piwakk) [inline frontend/src/models/repartitions.ts]: number?

File: frontend_legacy/src/app/app-routing.module.ts [hunk 10]
```diff
@@ -1,4 +1,4 @@
-import {NgModule} from '@angular/core';
+wimport {NgModule} from '@angular/core';
```
Comment (Piwakk) [inline frontend_legacy/src/app/app-routing.module.ts]: Pourquoi déranger les morts ?

File: frontend/src/models/polls.ts [hunk 11]
```diff
@@ -0,0 +1,28 @@
...
+}
+
+export interface Poll {
+    id: string;
+    choices: Choice[];
+    question: string;
+    state: PollState;
+    creationDateTime: Date;
+    publication_date: Date;
+    admin_comment: string;
```
Comment (Piwakk) [inline frontend/src/models/polls.ts]: HA HA!

File: frontend/src/models/repartitions.ts [hunk 12]
```diff
@@ -0,0 +1,42 @@
+import { User } from "./user";
+
+export enum CampaignStatus {
+    CLOSED = "CLOSED",
+    OPEN = "OPEN",
+    RESULTS = "RESULTS"
```
Comment (Piwakk) [inline frontend/src/models/repartitions.ts]: Still not changed

**Pull request merged**